On The Map/Controller/AddLocationViewController.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  AddLocationVC.swift
+//  On The Map
+//
+//  Created by Michal Hus on 2/23/20.
+//  Copyright © 2020 Michal Hus. All rights reserved.
+//
+
+import UIKit
+import MapKit
+import CoreLocation
+
+class AddLocationViewController: UIViewController {
+    
+    var coordinate: CLLocationCoordinate2D = CLLocationCoordinate2D()
+    var activityIndicator: UIActivityIndicatorView = UIActivityIndicatorView()
+    
+    @IBOutlet weak var location: UITextField!
+    @IBOutlet weak var link: UITextField!
+    
+    @IBAction func cancelOut(_ sender: Any) {
+        self.dismiss(animated: true, completion: nil)
+    }
+    
+    @IBAction func addLocation(_ sender: Any) {
+        
+        activityIndicator.center = self.view.center
+        activityIndicator.hidesWhenStopped = true
+        activityIndicator.style = UIActivityIndicatorView.Style.gray
+        view.addSubview(activityIndicator)
+        activityIndicator.startAnimating()
+        UIApplication.shared.beginIgnoringInteractionEvents()
+        
+        Client.getUserPublicInfo { (success, error) in
+            guard success, error == nil else {
+                self.errorAlertMessage(title: "Fetching User Data Failure", message: error!.localizedDescription)
+                return
+            }
+        }
+        
+        guard let location = location.text, location != ""  else {
+            errorAlertMessage(title: "Location Failure", message: "Please enter a locatin.")
+            return
+        }
+        
+        guard let link = link.text, link != "" else {
+            errorAlertMessage(title: "Link Failure", message: "Please enter a link.")
+            return
+        }
```

Comment:
getUserPublicInfo is asynchronous, but there is synchronous code that comes after it that will be run before `getUserPublicInfo` calls its completion block, which is a little confusing

Reply:
Something to talk about

Reply:
Let's talk about it and what would a good way of handling it.